rtrlib/rtr/packets.c

```diff
@@ -1226,11 +1357,33 @@ static int rtr_sync_receive_and_store_pdus(struct rtr_socket *rtr_socket)
 					retval = RTR_ERROR;
 					goto cleanup;
 				}
+								
+				aspa_update_table = lrtr_malloc(sizeof(struct aspa_table));
+				if (!aspa_update_table) {
+					RTR_DBG1("Memory allocation for aspa shadow table failed");
+					retval = RTR_ERROR;
+					goto cleanup;
+				}
+				if (aspa_table_cache_init(aspa_update_table, rtr_socket) != ASPA_SUCCESS) {
+					RTR_DBG1("Creation of aspa cache table failed");
+					rtr_change_socket_state(rtr_socket, RTR_ERROR_FATAL);
+					retval = RTR_ERROR;
+					goto cleanup;
+				}
+				aspa_cache_table = aspa_update_table;
 
 				RTR_DBG1("Shadow table created");
+				
+				/*
+				 X_update_table = copy rtr_socket->X_table if record->socket != rtr_socket;
```

Comment:
why is this code commented out ?

Reply:
I really really absolutely don't know.. The whole shadow table renaming just seems to be to have a variable outside the while loop, everything's weird here